### PR TITLE
Update upgrade instructions for accuracy and alignment through all versions

### LINKED ISSUE #3288: Upgrade Sensu Go instructions need further clarification
### Description

As of now, the Upgrade Sensu instructions simply state the following, for anyone who is upgrading to the latest version: 

```To upgrade to the latest version of Sensu Go, install the latest packages and restart the services.```

However, if the user is on `5.x`, there are additional steps, including running `sensu-backend upgrade`. This is not made clear on the Docs. 

### Category

This is an issue with:

- [x] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages

https://docs.sensu.io/sensu-go/latest/operations/maintain-sensu/upgrade/

### Corrections needed

The instructions on this page are broken down by versions, but they do not connect or build on one another. As it is now, the section `Upgrade to Sensu Go 6.4.0 from any previous version` does not state that any prior steps need to be done. For instance, if you are upgrading to `6.4` from` 5.x`, it needs to be made clear that the steps for each prior version upgrade must be followed first - or at the very least, the steps from `Upgrade to Sensu Go 6.0 from a 5.x deployment` (specifically, running the `sensu-backend upgrade` command) must first be followed before upgrading to` 6.4`.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Moves upgrade instructions to a step list at the top of the page
- Introduces first step in upgrade process that instructs users to read and follow upgrade instructions for all previous versions, starting with their current version and continuing up to the version they want to install
- Removes repetitive standard upgrade process steps from version-specific sections
- Adds references and links to standard upgrade process throughout as appropriate

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/3288

### CODE REVIEW

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md [hunk 1]
```diff
@@ -11,25 +11,67 @@ menu:
...
 sudo service sensu-backend restart
 {{< /code >}}
 
-To confirm the installed version, run `sensu-agent version`, `sensu-backend version`, and `sensuctl version`.
+6. Upgrade the Sensu backend:
+{{< code shell >}}
+sensu-backend upgrade
+{{< /code >}}
+
+7. Run a single upgrade command on one your Sensu backends to migrate the cluster:
```
Comment (mgibson323) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]: steps 6 and 7 are asking the user to run the same command - are they to run this command twice?

**Review by mgibson323 — CHANGES_REQUESTED** [review]
I have one question but other than that, this looks good as far as I understand the process. Thanks!

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]: No, that is just a mistake I forgot to fix when I combined the upgrade instructions from a couple sections. Thank you, I'll fix it now.

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md [hunk 2]
```diff
@@ -11,25 +11,67 @@ menu:
...
 {{< code shell >}}
 sudo service sensu-backend restart
 {{< /code >}}
 
-To confirm the installed version, run `sensu-agent version`, `sensu-backend version`, and `sensuctl version`.
+6. Upgrade the Sensu backend:
+{{< code shell >}}
+sensu-backend upgrade
+{{< /code >}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]:
```suggestion
```

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md (hunk 1 above)
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]:
```suggestion
6. Run a single upgrade command on one your Sensu backends to migrate the cluster:
```

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md [hunk 3]
```diff
@@ -11,25 +11,67 @@ menu:
...
+   To skip confirmation and immediately run the upgrade command, add the `--skip-confirm` flag:
+{{< code shell >}}
+sensu-backend upgrade --skip-confirm
+{{< /code >}}
+
+   {{% notice note %}}
+   **NOTE**: If you are deploying a new Sensu cluster rather than upgrading from a previous version, you do not need to run the `sensu-backend upgrade` command.
+{{% /notice %}}
+
+8. Enter `y` or `n` to confirm if you did *not* add the `--skip-confirm` flag in step 7.
```
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]:
```suggestion
7. Enter `y` or `n` to confirm if you did *not* add the `--skip-confirm` flag in step 6.
```

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md [hunk 4]
```diff
@@ -11,25 +11,67 @@ menu:
...
+Otherwise, skip this step.
+
+9. Wait a few seconds for the upgrade command to run.
```
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]:
```suggestion
8. Wait a few seconds for the upgrade command to run.
```

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md [hunk 5]
```diff
@@ -11,25 +11,67 @@ menu:
...
+
+   You may notice some inconsistencies in your entity list until the cluster finishes upgrading.
+Despite this, your cluster will continue to publish standard check requests and process events.
+
+   If you run the upgrade command more than once, it will not harm the cluster &mdash; you'll just see a response that the upgrade command has already been run.
+
+10. Confirm the installed version for the agent, backend, and sensuctl:
```
Comment (hillaryfraley) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md]:
```suggestion
9. Confirm the installed version for the agent, backend, and sensuctl:
```

**Review by echlebek — CHANGES_REQUESTED** [review]
So this is way more correct than before and should lead to a user eventually getting a properly-upgraded cluster. However, the typical upgrade path can be simpler than what is expressed here.

> Starting with your current version, read the version-specific information on this page and complete the instructions for each individual version up to the version you want to install.

This should be unnecessary unless debugging an upgrade that isn't succeeding. Most users can simply run `sensu-backend upgrade` once after upgrading to the newest version, and the tool will run all migrations that are necessary.

**Comment by hillaryfraley** [comment]
@echlebek that's very helpful -- thank you!

File: content/sensu-go/6.4/operations/maintain-sensu/upgrade.md [hunk 6]
```diff
@@ -11,25 +11,64 @@ menu:
...
+
+6. Enter `y` or `n` to confirm if you did *not* add the `--skip-confirm` flag in step 5.
+Otherwise, skip this step.
+
+7. Wait a few seconds for the upgrade command to run.
+
+   You may notice some inconsistencies in your entity list until the cluster finishes upgrading.
+Despite this, your cluster will continue to publish standard check requests and process events.
+
+   If you run the upgrade command more than once, it will not harm the cluster &mdash; you'll just see a response that the upgrade command has already been run.
```
Comment (echlebek) [inline content/sensu-go/6.4/operations/maintain-sensu/upgrade.md:54]: I'd like to add a sentence or two here.

"For some minor version upgrades, there will be no database-specific changes, so the `sensu-backend upgrade` tool will report that nothing has been upgraded. Check the release notes for the intermediate versions to see if this is expected or not.

**Review by echlebek — APPROVED** [review]
Thank Hillary!

**Pull request merged**